### PR TITLE
T18575

### PR DESCRIPTION
**Pull request by ahmedasaleh**
This ticket related to using charts with indicators in RTL language like Arabic. When touching the start or the end of the chart, two indicators appears.

### CODE REVIEW

**Comment by dinasayed** [comment]
This PR is related to https://bugs.dojotoolkit.org/ticket/18575

**Comment by dylans** [comment]
Thanks for this fix and test @ahmedasaleh . I would like to kindly suggest that any new tests be written as AMD rather than legacy dojo.require style code (ideally I'd actually ask that they be authored with Intern, but I know that dojox has not been converted to Intern yet, as it's a substantial undertaking to rewrite all of the tests).

**Comment by ahmedasaleh** [comment]
Noted @dylans will be taken into consideration in upcoming changes. Any feedback about the changes in _IndicatorElement.js  which is the core work

File: charting/action2d/_IndicatorElement.js [hunk 1]
```diff
@@ -233,6 +234,19 @@ define(["dojo/_base/lang", "dojo/_base/array", "dojo/_base/declare", "../plot2d/
 					break;
 				}
 			}
+			if(has("dojo-bidi")){
+				i = l - i;
+				r = data[i];
+				//I found by testing that the renderer uses the right side of the arrow pointer to calculate the X position
```
Comment (dylans) [inline charting/action2d/_IndicatorElement.js]: I'd probably change the explanation from first person to just explaining the issue, but that's just a minor style note on the comment.

File: charting/action2d/_IndicatorElement.js [hunk 2]
```diff
@@ -233,6 +234,19 @@ define(["dojo/_base/lang", "dojo/_base/array", "dojo/_base/declare", "../plot2d/
...
+				//that's should be fine for LTR but for RTL we need to use the left side, hence I added the arrow width
+				//1 to fix the error
+				if(cd[attr] > 1){
+					cd[attr] = l - cd[attr] + 1;
+					if(cd[attr] > l)
```
Comment (dylans) [inline charting/action2d/_IndicatorElement.js]: I'd feel more comfortable with a comment explaining the logic fully, as while it seems to work, I'm not sure the next person that looks at this code will understand why.

**Comment by dylans** [comment]
@ahmedasaleh comments added... the code change itself looks fine, but as noted, I'd feel more comfortable with a comment explaining what the code is doing, as I don't think it will be clear to the next person having to work through this code.

**Comment by ahmedasaleh** [comment]
@dylans any further comments on the code or feedback? will it be merged into the master?

**Comment by dylans** [comment]
Thanks... I'd suggest that the test example use AMD (I know that many of the existing tests still use old syntax, but any new tests should use AMD). Either way, we can land it this week.

**Comment by ahmedasaleh** [comment]
@dylans I modified the test file to use AMD, would you please check and let me know if you have further comments

File: charting/tests/mirror_touchIndicator.html [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+<script>
+	require([ "dojox/charting/Chart", "dojox/charting/axis2d/Default",
+			"dojox/charting/plot2d/StackedAreas",
+			"dojox/charting/action2d/TouchZoomAndPan",
+			"dojox/charting/action2d/TouchIndicator",
+			"dojox/charting/themes/Wetland", "dojo/ready" ], function(Chart,
+			Default, StackedAreas, TouchZoomAndPan, TouchIndicator, Wetland,
+			ready) {
+		ready(function() {
```
Comment (dylans) [inline charting/tests/mirror_touchIndicator.html]: ready isn't something that should be needed post Dojo 1.8. You should just be able to use dojo/domReady! and then work with the code directly inside this block.

File: charting/tests/mirror_touchIndicator.html [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+					"Series A",
+					[ 8, 7, 3, 2, 5, 7, 9, 10, 2, 10, 14, 16, 29, 13, 16, 15,
+							20, 19, 15, 12, 24, 20, 20, 26, 28, 26, 28, 14, 24,
+							29, 31, 35, 37, 31, 35, 37, 37, 36, 31, 30, 50, 49,
+							42, 46, 44, 40, 47, 43, 48, 47, 51, 52, 52, 51, 54,
+							57, 58, 50, 54, 51, 74, 68, 67, 62, 62, 65, 61, 66,
+							65, 62, 74, 78, 78, 77, 74, 62, 72, 74, 70, 78, 84,
+							83, 85, 86, 86, 89, 89, 85, 86, 86, 98, 73, 93, 91,
+							92, 92, 99, 93, 94, 92 ])
+			new dojox.charting.action2d.TouchZoomAndPan(c, "default", {
```
Comment (dylans) [inline charting/tests/mirror_touchIndicator.html]: should be new TouchZoomAndPan, not dojox.charting.action2d...

File: charting/tests/mirror_touchIndicator.html [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+				axis : "x",
+				enableScroll : false,
+				enableZoom : false
+			});
+			new dojox.charting.action2d.TouchIndicator(c, "default", {
```
Comment (dylans) [inline charting/tests/mirror_touchIndicator.html]: new TouchIndicator

**Comment by dylans** [comment]
@ahmedasaleh a few minor things to fix, and then this will be ready to land. Thanks!

**Comment by ahmedasaleh** [comment]
@dylans done

**Comment by dylans** [comment]
Thanks @ahmedasaleh! closed via b93b2e0

**Pull request closed without merging**